yt/data_objects/construction_data_containers.py

```diff
@@ -871,6 +871,12 @@ def _fill_sph_particles(self, fields):
 
         bounds, size = self._get_grid_bounds_size()
 
+        period = self.ds.coordinates.period.copy()
+        if hasattr(period, "in_units"):
```

Comment:
wait, why would this condition ever _not_ be checked ? can this array ever be a vanilla ndarray ?

Reply:
I'm not sure--this logic was copied over from the different coordinate handlers, which seem to think this is a possibility.

Reply:
Yes, it can be.

Reply:
Ok, this could probably be bullet-proofed but it's not the point here. Thanks for your prompt reply !

Reply:
At some point we also want to get to the point where it does not ever require conversion, IMO -- which is not the same as requiring it is vanilla ndarray.